examples/projects/zephyr/src/main.c

```diff
@@ -1,16 +1,29 @@
 #include <zephyr/kernel.h>
 #include <luos_engine.h>
+#include <robus_hal.h>
 #include <led.h>
 
-int main(void)
+
+static void luos_thread_loop(void *arg1, void *arg2, void *arg3);
+K_THREAD_DEFINE(luos_thread, 8192, luos_thread_loop, NULL, NULL, NULL, 8, 0, 0);
+
+static void luos_thread_loop(void *arg1, void *arg2, void *arg3)
 {
     Luos_Init();
     Led_Init();
+
     while (1)
-    {
+    {   
+        //Way to make Luos run in event driven mode 
+        RobusHAL_WaitNetworkActivity();
```

Comment:
We planned to create a macro on Luos for this!

Reply:
Will work on it 👍👨‍💻

Reply:
Please check also the implementation on Robus side, my initial idea was to defer a user defined callback, although a binary semaphore solves very well the issue of trigeering Luos loop upon a network activity.